lib/api_request.js

```diff
@@ -74,70 +70,30 @@ function get_hotel_info (hid, callback) {
-
-    var totalHits = 0;
-    A.mapLimit(
-      hids,
-      10,
-      function iteratee (hid, done) {
-        var _params = JSON.parse(JSON.stringify(params));
-        _params.hotelIds = hid;
-        var options = make_options(_params);
-        var startedAt = Date.now();
-        var req = 'https://' + options.host + options.path;
-        http_request(options, function (err, data) {
-          if (!err && data.result && Array.isArray(data.result) && data.result.length > 0) {
-            var pkg = [data.result.sort(sort_by_price_asc)[0]]; // get cheapest package
-            AwsHelper.log.trace({ delta: Date.now() - startedAt }, 'Amount of time to fetch packages');
-            var startedAt2 = Date.now();
-            get_hotel_info(hid, function (err, hotel_info) {
-              /* istanbul ignore if */
-              if (err) {
-                AwsHelper.log.trace({get_hotel_info_error: err, hotelId: hid, request: req, date: new Date()}, 'Error fetching hotels');
-                AwsHelper.log.warn({ err: err, hotelId: hid, request: req }, 'Unable to get hotel info');
-                return done(null, null);
-              }
-
-              AwsHelper.log.trace({ delta: Date.now() - startedAt2 }, 'Amount of time to fetch hotel info');
-
-              // format one package result at a time:
-              var records = mapper.map_ne_result_to_graphql(pkg, hotel_info);
-              body.items = records;
-              AwsHelper.log.trace({'api_request': req, 'hits': data.totalHits, 'date': new Date()}, 'Api request: ' + data.totalHits + ' hits');
-              ee.emit('result', body);
-              totalHits += data.totalHits;
-
-              return done(null, records[0]);
-            });
-          } else {
-            AwsHelper.log.trace({'api_request_error': err, 'hotelId': hid, request: req, date: new Date()}, 'Api request error');
-            return done(null, null);
-          }
-        });
-      },
-      function (_err, results) {
-        assert(!_err); // err is ALWAYS null (never callback with actual error)
-        AwsHelper.log.info({ 'results': totalHits }, 'Package Results');
-        results = results.filter(function (a) { return a !== null; });
-        return callback(_err, { result: results, totalHits: totalHits });
+  params = clone(params);
+  const options = make_options(params);
+  AwsHelper.log.trace({ req: options, hotelId: params.hotelId }, 'Fetching package info');
+  http.request(options, function (err, data) {
+    if (!err && data.result && Array.isArray(data.result) && data.result.length > 0) {
+      const pkg = data.result.sort(sort_by_price_asc)[0]; // get cheapest package
+      get_hotel_info(params, function (err, hotel) {
+        /* istanbul ignore if */
+        if (err) {
+          AwsHelper.log.warn({ err: err, hotelId: params.hotelId, request: options }, 'Unable to get hotel info');
+          callback();
```

Comment:
Well... we don't want errors in any one particular hotel's api request pipeline to cause errors in the full result set (fault tolerance again), so we certainly need to ignore this error at some level. (*Note - [it's ignored in the current code](https://github.com/numo-labs/lambda-ne-dynamic-package-provider/blob/814cb4d38645b48eb27e3b93c5c9d11246ff3006/lib/api_request.js#L113)*)

There is an argument to be made that this module as a standalone component should throw errors as usual though, and that they should be handled at the call site. If I were building from scratch I'd probably do that, it's just as a progression from the previous iteration that it's the way it is.

tl;dr - probably yes, I'll update it.